spec/services/link_check_report/update_service_spec.rb

```diff
@@ -0,0 +1,102 @@
+require "spec_helper"
+
+RSpec.describe LinkCheckReport::UpdateService do
+  let(:link_check_report) do
+    LinkCheckReport.new(
+      batch_id: 1,
+      status: "in progress",
+      link_reportable_type: "manual",
+      link_reportable_id: 2
+    )
+  end
+
+  let(:link_1) do
+    Link.new(
+      link_check_report: link_check_report,
+      uri: "http://www.example.com",
+      status: "pending"
+    )
+  end
+
+  let(:link_2) do
+    Link.new(
+      link_check_report: link_check_report,
+      uri: "http://www.gov.uk",
+      status: "pending"
+    )
+  end
+
+  let(:completed_at) { Time.now }
+
+  let(:payload) do
+    {
+      status: 'complete',
+      completed_at: completed_at,
+      links: links_payload
+    }.with_indifferent_access
+  end
+
+  let(:links_payload) do
+    [{
+      uri: "http://www.example.com",
+      status: "ok",
+      checked: completed_at.try(:iso8601),
+      problem_summary: nil,
+      suggested_fix: nil
+    }, {
+      uri: "http://www.gov.com",
+      status: "broken",
+      checked: completed_at.try(:iso8601),
+      problem_summary: "Page Not Found",
+      suggested_fix: "Contact site administrator"
+    }]
+  end
+
+  subject do
+    described_class.new(report: link_check_report, payload: payload)
+  end
+
+  before do
+    allow(link_check_report).to receive(:update!)
+    allow(Link).to receive(:find).with(link_check_report: link_check_report, uri: "http://www.example.com").and_return(link_1)
```

Comment:
Is it better to use Factory Bot here to have these available?

Reply:
Updated to use factory @thomasleese :)

Reply:
Could we use `create` here so we get real objects and avoid having to stub out the database query methods?

Reply:
Happy to, but I tried to keep these tests symmetrical to the other [update tests](https://github.com/alphagov/manuals-publisher/blob/master/spec/services/section/update_service_spec.rb) within the repo.

Reply:
The history of this app means it's written in a way that's not very conventional Rails, I wouldn't worry about matching the other tests.